versionist.conf.js

```diff
@@ -1,32 +1,62 @@
-module.exports = {
+var _ = require('lodash');
 
-  subjectParser: 'angular',
+lowerKeys = (obj) => {
+  return _.transform(obj, (result, val, key) => {
+    result[key.toLowerCase()] = val;
+  });
+};
 
+module.exports = {
+  // This setup allows the editing and parsing of footer tags to get version and type information,
+  // as well as ensuring tags of the type 'v<major>.<minor>.<patch>' are used.
+  // It increments in a semver compatible fashion and allows the updating of NPM package info.
+  editChangelog: true,
+  parseFooterTags: true,
   getGitReferenceFromVersion: 'v-prefix',
+  incrementVersion: 'semver',
+  updateVersion: 'npm',
 
+  // Always add the entry to the top of the Changelog, below the header.
   addEntryToChangelog: {
     preset: 'prepend',
     fromLine: 5
   },
 
-  updateVersion: 'npm',
-
+  // Only include a commit when there is a footer of 'Change-Type'.
+  // Ensures commits which do not up versions are not included.
+  // Commit messages without a relevant footer will not be included in the CHANGELOG.
   includeCommitWhen: (commit) => {
-    return commit.footer['Changelog-Entry'];
+    commit.footer = lowerKeys(commit.footer);
+    return !!commit.footer['change-type'];
   },
 
+  // Determine the type from 'change-yype:' tag.
+  // Should no explicit change type be made, then no changes are assumed.
   getIncrementLevelFromCommit: (commit) => {
-    return commit.footer['Change-Type'];
+    commit.footer = lowerKeys(commit.footer);
+    if (commit.footer['change-type']) {
+      return commit.footer['change-type'].trim();
+    }
+  },
+
+  // If a 'changelog-entry' tag is found, use this as the subject rather than the
+  // first line of the commit.
+  transformTemplateData: (data) => {
+    data.commits.forEach((commit) => {
+      commit.footer = lowerKeys(commit.footer);
+      if (commit.footer['changelog-entry']) {
+        commit.subject = commit.footer['changelog-entry'];
+      }
+    });
+
+    return data;
   },
 
   template: [
-    '## {{version}} - {{moment date "Y-MM-DD"}}',
+    '# v{{version}}',
```

Comment:
Why remove the date? Always helpful to know how stale / fresh a piece of code is (even if it's on a separate line rather than the same line as the version string).